### PR TITLE
(QENG-1847) changed internal host path attribute usage to host.puppet me...

### PR DESCRIPTION
**Pull request by kevpl**
...thod calls where possible

### CODE REVIEW

**Comment by kevpl** [comment]
@puppetlabs-jenkins retest this please

**Comment by kevpl** [comment]
@puppetlabs-jenkins retest this please

**Comment by joshcooper** [comment]
The failures are caused because the puppet package is not creating the `/usr/share/puppet/modules` system module directory leading to:

```
Cannot create /usr/share/puppet/modules/crick; parent directory /usr/share/puppet/modules does not exist
```

This is filed as https://tickets.puppetlabs.com/browse/PUP-4001

File: lib/beaker/dsl/helpers.rb [hunk 1]
```diff
@@ -553,7 +553,7 @@ def with_puppet_running_on host, conf_opts, testdir = host.tmpdir(File.basename(
         end
 
         begin
-          backup_file = backup_the_file(host, host['puppetconfdir'], testdir, 'puppet.conf')
+          backup_file = backup_the_file(host, host.puppet['confdir'], testdir, 'puppet.conf')
```
Comment (joshcooper) [inline lib/beaker/dsl/helpers.rb]: I believe `with_puppet_running_on` is only ever called on the master node, in which case, I think we always want to do `host.puppet('master')['confdir']`.

Currently the agent and master share the confdir (so it shouldn't matter), but the confdir may be different in the future, e.g. the master's vardir is changing https://tickets.puppetlabs.com/browse/SERVER-369

Comment (anodelman) [inline lib/beaker/dsl/helpers.rb]: If 'with_puppet_running_on' can only be called by the master we should put in a check for that and fail out otherwise.

**Comment by kevpl** [comment]
review changes added

**Comment by rji** [comment]
retest this please

**Comment by rji** [comment]
@anodelman,@kevpl - All cells are green except the expected failure from FOSS Puppet on Debian.

**Pull request merged**